web/src/engine/osk/src/input/gestures/specsForKeyboard.ts

```diff
@@ -0,0 +1,349 @@
+  // if `null`, assume a no-flick keyboard (assuming our default layout has no flicks)
+
+  // Idea:  if we want to get fancy, we could detect if the keyboard even _supports_ some of
+  // the less common gestures and just... not include the model if it doesn't.  Should only
+  // do that if it's computationally "cheap", though.
+  return {
+    gestures: [
+      // TODO:  some, if not all, will probably utilize methods, rather than constant definitions.
+      // But, this should be fine for a first-pass integration attempt.
+      LongpressModel,
+      MultitapModel,
+      SimpleTapModel,
+      SubkeySelectModel,
+      ModipressStartModel,
+      ModipressEndModel
+    ],
+    sets: {
+      default: [LongpressModel.id, SimpleTapModel.id, ModipressStartModel.id],
+      modipress: [LongpressModel.id, SimpleTapModel.id], // no nested modipressing
+      none: []
+    }
+  }
+}
+
+// #region Definition of models for paths comprising gesture-stage models
+
+type ContactModel = specs.ContactModel<KeyElement>;
+
+export const InstantContactRejectionModel: ContactModel = {
+  itemPriority: 0,
+  pathResolutionAction: 'reject',
+  pathModel: {
+    evaluate: (path) => 'resolve'
+  }
+}
+
+export const InstantContactResolutionModel: ContactModel = {
+  itemPriority: 0,
+  pathResolutionAction: 'resolve',
+  pathModel: {
+    evaluate: (path) => 'resolve'
+  }
+}
+
+export const LongpressDistanceThreshold = 10;
+export const MainContactLongpressSourceModel: ContactModel = {
+  itemChangeAction: 'reject',
+  itemPriority: 0,
+  pathResolutionAction: 'resolve',
+  timer: {
+    duration: 500,
+    expectedResult: true
+  },
+  pathModel: {
+    evaluate: (path) => {
+      const stats = path.stats;
+      if(stats.rawDistance > LongpressDistanceThreshold) {
+        return 'reject';
+      }
+
+      if(path.isComplete) {
+        return 'reject';
+      }
+    }
+  }
+};
+
+export const LongpressFlickDistanceThreshold = 6;
+export const MainContactLongpressSourceModelWithShortcut: ContactModel = {
+  ...MainContactLongpressSourceModel,
+  pathModel: {
+    evaluate: (path) => {
+      const stats = path.stats;
+
+      // Adds up-flick support!
+      if(stats.rawDistance > LongpressFlickDistanceThreshold && stats.cardinalDirection == 'n') {
+        return 'resolve';
+      }
+
+      return MainContactLongpressSourceModel.pathModel.evaluate(path);
+    }
+  }
+}
+
+export const ModipressContactStartModel: ContactModel = {
+  itemPriority: -1,
+  pathResolutionAction: 'resolve',
+  pathModel: {
+    // Consideration of whether the underlying item supports the corresponding
+    // gesture will be handled elsewhere.
+    evaluate: (path) => 'resolve'
+  }
+}
+
+export const ModipressContactEndModel: ContactModel = {
+  itemPriority: -1,
+  itemChangeAction: 'resolve',
+  pathResolutionAction: 'resolve',
+  pathModel: {
+    evaluate: (path) => {
+      if(path.isComplete) {
+        return 'resolve';
+      }
+    }
```

Comment:
```suggestion
    evaluate: path => path.isComplete ? 'resolve' : undefined,
```

Is `undefined` really what we should be returning otherwise?

Reply:
https://github.com/keymanapp/keyman/blob/275202c69f9be9d5a78740252ede644a5e192256/common/web/gesture-recognizer/src/engine/headless/gestures/specs/pathModel.ts#L6-L15

Looks like I need to update the doc-comment (`TrackedPath` => `GesturePath`), but the rest of this holds.

https://github.com/keymanapp/keyman/blob/275202c69f9be9d5a78740252ede644a5e192256/common/web/gesture-recognizer/src/engine/headless/gestures/matchers/pathMatcher.ts#L131

If you think it feels better, the actual check is "falsy"-based, so `null` should work as well.